### PR TITLE
Remove events parameter from Span creation

### PR DESCRIPTION
**Pull request by fbogsany**
Implements https://github.com/open-telemetry/opentelemetry-specification/pull/252

### CODE REVIEW

**Comment by fbogsany** [comment]
I'm tempted to re-expose `Link` and `Event` in the API, remove `create_event` and `create_link`, and trim both the arrays and attributes in `TraceConfig.trim_links` and `TraceConfig.trim_events`.

WDYT @mwear? I'm happy to do that refactoring in this PR.

**Comment by mwear** [comment]
@fbogsany I'd be fine with that. It'd make us less of a special snowflake regarding links and events. I'm curious what your reasoning for reintroducing them is.

**Comment by fbogsany** [comment]
> I'm curious what your reasoning for reintroducing them is.

It simplifies the codebase. I can still coalesce the trimming in the `TraceConfig` class.

**Comment by fbogsany** [comment]
I've moved `Event` and `Link` back to the API. I think it simplifies both the API and the SDK, and should be more efficient in the common cases. The `TraceConfig` trimming code is a bit more complex, but I think the tradeoff is worthwhile for the simplicity & concreteness of the API.

File: sdk/lib/opentelemetry/sdk/trace/config/trace_config.rb [hunk 1]
```diff
@@ -68,78 +68,94 @@ def initialize(sampler: DEFAULT_SAMPLER,
...
-            links.shift(excess) if excess.positive?
-            nil
+          # Also trims attributes of each {OpenTelemetry::Trace::Link} to have
+          # no more than {max_attributes_per_link} entries.
+          #
+          # @param [Array<OpenTelemetry::Trace::Link>] links Array of
+          #   {OpenTelemetry::Trace::Link}s to trim. May be nil.
+          # @return [Array<OpenTelemetry::Trace::Link>] frozen slice of links
+          #   param. May be nil.
+          def trim_links(links) # rubocop:disable Metrics/AbcSize
```
[the remark this answers is not in the document]
Comment (fbogsany) [reply sdk/lib/opentelemetry/sdk/trace/config/trace_config.rb]: Very good point. I'll move them to live with their callers.

[the remark this answers is not in the document]
Comment (fbogsany) [reply sdk/lib/opentelemetry/sdk/trace/config/trace_config.rb]: Done.

**Comment by fbogsany** [comment]
> I just have the comment about some of the helper methods in TraceConfig. We can discuss and see if we like them where they are, if we should find them a new home (but possibly not now), or if we'd like to do something in this PR.

I moved them to `Span`. It's the right place for them.

**Pull request merged**